### PR TITLE
Auth tokens

### PR DESCRIPTION
**Pull request by ericnewcomer**
Add new property ```APIToken.role```. 

* Migrate current tokens to have role based on user group for org
* Apply role when generating new tokens
* Require role when authenticating against the api
* Take role into account with authenticating with token

### CODE REVIEW

File: temba/api/models.py [hunk 1]
```diff
@@ -479,6 +479,7 @@ class APIToken(models.Model):
     user = models.ForeignKey(User, related_name='api_tokens')
     org = models.ForeignKey(Org, related_name='api_tokens')
     created = models.DateTimeField(auto_now_add=True)
+    role = models.CharField(max_length=1, null=False)
```
Comment (nicpottier) [inline temba/api/models.py]: I don't know man, this seems super jank.. why not a foreign key to a UserGroup? Just seems to be screaming for badness to occur if UserGroups are created down the line with similar starting letters.

File: temba/api/views.py [hunk 2]
```diff
@@ -71,20 +72,35 @@ def webhook_status_processor(request):
...
                 return False
 
-            org = request.user.get_org()
-            if org:
+            group = None
+
+            # try to determine group from api token
+            if request.auth:
+                if request.auth.role:
+                    group = Group.objects.filter(name__startswith=request.auth.role).first()
```
Comment (nicpottier) [inline temba/api/views.py]: Oy!

Comment (ericnewcomer) [inline temba/api/views.py]: ya, saw that coming ;)

**Comment by ericnewcomer** [comment]
Alright, less cute version ready for review.

**Comment by norkans7** [comment]
Looks good

**Pull request merged**